backend/PennCourses/settings/base.py

```diff
@@ -181,3 +182,10 @@
 }
 
 STATS_WEBHOOK = os.environ.get("STATS_WEBHOOK", None)
+
+S3 = boto3.client(
+    "s3",
+    region_name="us-east-1",
+    aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID"),
+    aws_secret_access_key=os.environ.get("AWS_SECRET_ACCESS_KEY"),
```

Comment:
What happens if you remove these lines? In the future, we're planning on moving away from providing explicit access/secret keys in production and I'm pretty sure boto3 will automatically find the credentials, but I'm not 100% sure

Reply:
Boto3 will look in several locations when searching for credentials. The mechanism in which Boto3 looks for credentials is to search through a list of possible locations and stop as soon as it finds credentials. The order in which Boto3 searches for credentials is:

1. Passing credentials as parameters in the boto.client() method
2. Passing credentials as parameters when creating a Session object
3. Environment variables
4. Shared credential file (~/.aws/credentials)
5. AWS config file (~/.aws/config)
6. Assume Role provider
7. Boto2 config file (/etc/boto.cfg and ~/.boto)
8. Instance metadata service on an Amazon EC2 instance that has an IAM role configured.

[https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html)

Reply:
So are we good to omit those kwargs / assume the creds will be found elsewhere? Also, maybe we want to move the boto3 client instantiation to production.py if we won't be using S3 in the code in dev at all?

Reply:
Yep, environment variables will be set in production. I think moving the config to production makes sense, but that would break the endpoints in dev. Would need to find a fix/workaround for it somehow. I also just realized that the load pcr data management command uses S3. Do we have two different sets of config for boot in this repo?